command_go120.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013-2023 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !go1.21
+// +build !go1.21
+
+package cobra
+
+import (
+	"os"
+	"strings"
+)
+
+// based on golang.org/x/mod/internal/lazyregexp: https://cs.opensource.google/go/x/mod/+/refs/tags/v0.19.0:internal/lazyregexp/lazyre.go;l=66
+var inTest = len(os.Args) > 0 && strings.HasSuffix(strings.TrimSuffix(os.Args[0], ".exe"), ".test")
```

Comment:
So this will break programs using Cobra with go < 1.21 which are named `<something>.test`.

For such a very rare case, what is the workaround?

Reply:
Hm, right. Bit on the  fence; should  that  be considered  _too much_ of a corner-case? If we don't want any risk at all, then we  could decide to keep the current, partially broken (e.g. not working on Windows), state for old go versions, and only apply the improvements to current versions of go.

I guess it depends  if we expect  users that are on EOL versions of Go  _do_ update dependencies  to "latest". 😅

Reply:
If this situation ever comes up at all I would just like to have a way for that program to address the issue instead of no longer being able to upgrade cobra anymore. 

I think such a program could do one of two things:
1. Upgrade to go >= 1.21, or
2. call `rootCmd.SetArgs(os.Args[1:])`

Do you agree?

Reply:
I tested this approach and it works, for either using go > 1.21 or calling `rootCmd.SetArgs(os.Args[1:])`